app/routes/labs/parallax/ParallaxImage.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useEffect, useState } from 'react'
+
+// Define an interface for the component's props
+interface ParallaxImageProps {
+  src: string // URL of the image
+  alt: string // Description of the image
+}
+
+function ParallaxImage({ src, alt }: ParallaxImageProps) {
+  const [offset, setOffset] = useState(0)
+
+  const handleScroll = () => {
+    const newOffset = window.pageYOffset
+    setOffset(newOffset)
+  }
+
+  useEffect(() => {
+    window.addEventListener('scroll', handleScroll)
+    return () => window.removeEventListener('scroll', handleScroll)
+  }, [])
```

Comment:
Does [usehooks-ts](https://usehooks-ts.com) supply any React hooks for watching updates to the scroll position? We already have a dependency on it.

Reply:
Yes, useEventListener from usehooks-ts let me accomplish the same thing, so I've replaced the code.